bin/node/runtime/src/lib.rs

```diff
@@ -2142,22 +2142,15 @@ impl_runtime_apis! {
 			impl baseline::Config for Runtime {}
 			impl pallet_nomination_pools_benchmarking::Config for Runtime {}
 
-			let whitelist: Vec<TrackedStorageKey> = vec![
-				// Block Number
-				hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac").to_vec().into(),
-				// Total Issuance
-				hex_literal::hex!("c2261276cc9d1f8598ea4b6a74b15c2f57c875e4cff74148e4628f264b974c80").to_vec().into(),
-				// Execution Phase
-				hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef7ff553b5a9862a516939d82b3d3d8661a").to_vec().into(),
-				// Event Count
-				hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef70a98fdbe9ce6c55837576c60c7af3850").to_vec().into(),
-				// System Events
-				hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef780d41e5e16056765bc8461851072c9d7").to_vec().into(),
-				// System BlockWeight
-				hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef734abf5cb34d6244378cddbf18e849d96").to_vec().into(),
-				// Treasury Account
-				hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da95ecffd7b6c0f78751baa9d281e0bfa3a6d6f646c70792f74727372790000000000000000000000000000000000000000").to_vec().into(),
-			];
+			use frame_support::traits::WhitelistedStorageKeys;
+			let mut whitelist: Vec<TrackedStorageKey> = AllPalletsWithSystem::whitelisted_storage_keys();
+
+			// Treasury Account (manually whitelist this for now since no
+			// accessible storage definition we can attach the
+			// benchmarking(cached) attr macro to)
+			// TODO: figure out a workaround for this that doesn't involve
+			// manually hard-coding
+			whitelist.push(hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da95ecffd7b6c0f78751baa9d281e0bfa3a6d6f646c70792f74727372790000000000000000000000000000000000000000").to_vec().into());
```

Comment:
We can 

1. manually implement `WhitelistedStorageKeys` for `Pallet` in `pallet-treasury` instead of using the macro. 

2. Or, not sure if it works, something along the lines of: 

```
trait WhitelistedStorageKeys {
  fn manually_whitelisted_keys(key: Vec<u8>) {}
  fn whitelisted_storage_items() {}
  fn final_whitelisted_keys() -> Vec<Vec<u8>> { 
    // combine the above two
  } 
```

and then we can do this hardcoding inside the pallet, not here.


At the minimum, we can try and use a combination of `System::Accounts::hased_key_for(Treasury::account())` here to remove hardcoded bytes.

Reply:
The problem with 1. is we immediately get a `conflicting implementation for `pallet::Pallet<_>`` error because we automatically implement `WhitelistedStorageKeys` for all pallets. We could do an additional attribute macro that marks a pallet such that we do not generate a default implementation of `WhitelistedStorageKeys` for it, allowing the pallet to define its own implementation. This seems a bit out of scope for what we set out to do, however, so I'd be for doing this in a follow-up PR and resolving this simply for now by just being neater by using `System::Accounts::hased_key_for(Treasury::account())`.

Thoughts?

Reply:
Also I keep getting compiler errors trying to do `System::Accounts::hased_key_for(Treasury::account())` -- do you know what the exact call would be from `bin/node/runtime/src/lib.rs:2153`?

Reply:
in particular I'm not sure how to import `System::Accounts` in this context

Reply:
What error do you get?

Reply:
Accounts is a public type in `frame_system::pallet` and is an alias to `SorageMap`:
https://paritytech.github.io/substrate/master/frame_system/pallet/type.Account.html# 
https://paritytech.github.io/substrate/master/frame_support/storage/types/struct.StorageMap.html#method.hashed_key_for

Reply:
The prefix `26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9` is `xx("System")++xx("Account")` the rest is probably some `Blake2_128Concat` result.

Reply:
I get something like this:
```
  error[E0433]: failed to resolve: could not find `Accounts` in `pallet`
      --> /home/sam/substrate/bin/node/runtime/src/lib.rs:2153:41
       |
  2153 |             whitelist.push(frame_system::pallet::Accounts::hashed_key_for(Treasury::account_id()).into());
       |                                                  ^^^^^^^^
       |
      ::: /home/sam/substrate/frame/system/src/lib.rs:551:5
       |
  551  |     pub type Account<T: Config> = StorageMap<
       |     --------------------------- similarly named type alias `Account` defined here
       |
  help: a type alias with a similar name exists
       |
  2153 |             whitelist.push(frame_system::pallet::Account::hashed_key_for(Treasury::account_id()).into());
       |                                                  ~~~~~~~
  help: consider importing this type alias
       |
  25   | use pallet_indices::Accounts;
       |
  help: if you import `Accounts`, refer to it directly
       |
  2153 -             whitelist.push(frame_system::pallet::Accounts::hashed_key_for(Treasury::account_id()).into());
  2153 +             whitelist.push(Accounts::hashed_key_for(Treasury::account_id()).into());
       |

```

Will try a few variations and report back

Reply:
And then if I try the `pallet_indices` way, I get some type stuff I'm not 100% sure how to resolve:
```
  error[E0284]: type annotations needed
      --> /home/sam/substrate/bin/node/runtime/src/lib.rs:2153:19
       |
  2153 |             whitelist.push(pallet_indices::Accounts::hashed_key_for(Treasury::account_id()).into());
       |                            ^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `T` declared on the type alias `Accounts`
       |
       = note: cannot satisfy `<_ as pallet_indices::Config>::AccountIndex == _`
```

Reply:
All set now I figured it out. Ended up using:

```rust
frame_system::Account::<Runtime>::hashed_key_for(Treasury::account_id())
```

And I was able to confirm this works properly. Thanks again for pointing me in the right direction!